pkg/armhelpers/deploymentError.go

```diff
@@ -0,0 +1,110 @@
+package armhelpers
+
+import (
+	"encoding/json"
+
+	"github.com/Azure/acs-engine/pkg/api"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/sirupsen/logrus"
+)
+
+// Error is the OData v4 format, used by the RPC and
+// will go into the v2.2 Azure REST API guidelines
+type Error struct {
+	Code    string  `json:"code"`
+	Message string  `json:"message"`
+	Target  string  `json:"target,omitempty"`
+	Details []Error `json:"details,omitempty"`
+}
+
+// ErrorResponse  defines Resource Provider API 2.0 Error Response Content structure
+type ErrorResponse struct {
+	Body Error `json:"error"`
+}
+
+// Error implements error interface to return error in json
+func (e *ErrorResponse) Error() string {
+	return e.Body.Error()
```

Comment:
is this recursive at all?  I remember in C# there were errors stuffed inside other errors, and turtles all the way down.

Reply:
yes, Error.Error() is recursive.